ApplicationLibCode/ProjectDataModel/Seismic/RimSeismicSection.cpp

```diff
@@ -505,18 +543,62 @@ QList<caf::PdmOptionItemInfo> RimSeismicSection::calculateValueOptions( const ca
 
         if ( view != nullptr ) RimTools::seismicDataOptionItems( &options, view->domainBoundingBox() );
     }
+    else if ( fieldNeedingOptions == &m_wellPath )
+    {
+        RimTools::wellPathOptionItems( &options );
+    }
 
     return options;
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimSeismicSection::updateTextureSectionFromPoints( std::vector<cvf::Vec3d> points, double zmin, double zmax )
+{
+    const int pointCount = (int)points.size();
+    if ( pointCount < 2 )
+    {
+        m_texturedSection->resize( 0 );
+        return;
+    }
+
+    bool valid = m_texturedSection->partsCount() == ( pointCount - 1 );
+    if ( !valid ) m_texturedSection->resize( pointCount - 1 );
```

Comment:
Use `m_texturedSection->resize` without checking size, as resize with same size is a noop.

Reply:
Uses valid flag later.